### Installing YOLO

In [ ]:
!pip install ultralytics

### MODEL TRAINING ON CUSTOM DATASET


In [1]:
# !nvidia-smi
import torch

print("CUDA Available:", torch.cuda.is_available())
print("GPU Name:", torch.cuda.get_device_name(0) if torch.cuda.is_available() else "No GPU")

CUDA Available: True
GPU Name: NVIDIA GeForce RTX 4050 Laptop GPU


In [2]:
import torch

print("Torch version:", torch.__version__)
print("CUDA available:", torch.cuda.is_available())
print("GPU:", torch.cuda.get_device_name(0) if torch.cuda.is_available() else "No GPU")

Torch version: 2.7.1+cu118
CUDA available: True
GPU: NVIDIA GeForce RTX 4050 Laptop GPU


In [ ]:
from ultralytics import YOLO

model = YOLO("yolov8n.pt")

model.train(
    data="D:/TE/Internship/code/data/dataset/v1_4000_images/dataset.yaml",
    epochs=50,
    imgsz=640,
    batch=16,
    device=0  # Uses RTX 4050
)

Ultralytics 8.4.15  Python-3.10.8 torch-2.7.1+cu118 CUDA:0 (NVIDIA GeForce RTX 4050 Laptop GPU, 6140MiB)
engine\trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=D:/TE/Internship/code/data/dataset/dataset.yaml, degrees=0.0, deterministic=True, device=0, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=50, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolov8n.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=train, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap

ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2, 3, 4])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x00000192BAD64880>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
    

<!-- ## Final Validation Results (After 50 Epochs)

| Metric      | Value |
|------------|--------|
| Precision  | 0.806  |
| Recall     | 0.672  |
| mAP50      | 0.762  |
| mAP50-95   | 0.553  |

## Per-Class mAP50

| Class     | mAP50 |
|-----------|--------|
| Bear      | 0.763 |
| Deer      | 0.792 |
| Elephant  | 0.817 |
| Monkey    | 0.866 |
| Tiger     | 0.572 | -->


## Final Validation Results (After 50 Epochs)

| Metric      | Value |
|-------------|-------|
| **Precision**  | 0.806 |
| **Recall**     | 0.672 |
| **mAP50**      | 0.762 |
| **mAP50-95**   | 0.553 |

## Per-Class mAP50

| Class     | mAP50 |
|-----------|-------|
| **Bear**      | 0.763 |
| **Deer**      | 0.792 |
| **Elephant**  | 0.817 |
| **Monkey**    | 0.866 |
| **Tiger**     | 0.572 |

# For 10000 Images

In [3]:
from ultralytics import YOLO

model = YOLO("yolo26n.pt")

model.train(
    data="D:/TE/Internship/code/data/dataset/v2_10000_images/dataset.yaml",
    epochs=50,
    imgsz=640,
    batch=16,
    device=0  # Uses RTX 4050
)

Ultralytics 8.4.15  Python-3.10.8 torch-2.7.1+cu118 CUDA:0 (NVIDIA GeForce RTX 4050 Laptop GPU, 6140MiB)
engine\trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=D:/TE/Internship/code/data/dataset/v2_10000_images/dataset.yaml, degrees=0.0, deterministic=True, device=0, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=50, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo26n.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=train, nbs=64, nms=False, opset=None, optimize=False, optimiz

ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x00000180E47FA980>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0

# YOLO Model Evaluation Results

**Model:** YOLO26n  
**Dataset:** Wildlife Dataset (10,000 Images)  
**Validation Images:** 2000  
**Total Instances:** 2907  
**Framework:** Ultralytics YOLO  
**GPU:** NVIDIA RTX 4050 Laptop GPU  

---

## Overall Metrics

| Metric | Value |
|------|------|
| Precision | **0.792** |
| Recall | **0.713** |
| mAP@0.5 | **0.777** |
| mAP@0.5:0.95 | **0.609** |

---

## Per-Class Performance

| Class | Images | Instances | Precision | Recall | mAP50 | mAP50-95 |
|------|------|------|------|------|------|------|
| Squirrel | 180 | 187 | 0.856 | 0.898 | 0.926 | 0.770 |
| Crocodile | 61 | 81 | 0.654 | 0.642 | 0.642 | 0.487 |
| Lizard | 191 | 220 | 0.834 | 0.732 | 0.788 | 0.513 |
| Elephant | 171 | 367 | 0.846 | 0.687 | 0.760 | 0.525 |
| Leopard | 66 | 75 | 0.768 | 0.705 | 0.789 | 0.629 |
| Cheetah | 51 | 61 | 0.722 | 0.639 | 0.720 | 0.588 |
| Deer | 215 | 368 | 0.805 | 0.734 | 0.805 | 0.613 |
| Giraffe | 100 | 150 | 0.823 | 0.780 | 0.829 | 0.657 |
| Snake | 125 | 133 | 0.830 | 0.697 | 0.791 | 0.607 |
| Monkey | 201 | 279 | 0.801 | 0.794 | 0.826 | 0.591 |
| Fox | 55 | 65 | 0.705 | 0.708 | 0.749 | 0.646 |
| Kangaroo | 58 | 83 | 0.835 | 0.731 | 0.781 | 0.642 |
| Lion | 122 | 170 | 0.838 | 0.700 | 0.801 | 0.637 |
| Tiger | 95 | 123 | 0.893 | 0.732 | 0.856 | 0.700 |
| Zebra | 70 | 134 | 0.787 | 0.769 | 0.812 | 0.658 |
| Rhinoceros | 91 | 127 | 0.938 | 0.836 | 0.919 | 0.819 |
| Bear | 31 | 32 | 0.697 | 0.531 | 0.590 | 0.537 |
| Panda | 80 | 117 | 0.803 | 0.661 | 0.780 | 0.544 |
| Hippopotamus | 24 | 46 | 0.632 | 0.696 | 0.713 | 0.569 |
| Otter | 51 | 89 | 0.766 | 0.588 | 0.671 | 0.453 |

---

## Inference Speed

| Stage | Time per Image |
|------|------|
| Preprocess | 1.0 ms |
| Inference | 2.6 ms |
| Postprocess | 0.1 ms |

---

## Results Directory
runs/10000_images_model_evaluation/detect/val10/

Contains:
- Confusion Matrix
- Precision–Recall Curve
- F1 Curve
- Precision Curve
- Recall Curve